### PR TITLE
APPLE-63: New file upload for config files

### PR DESCRIPTION
**Pull request by tylermachado**
This eliminates the individual inputs for API config credentials, and replaces them with a file upload (or text area) for a single .papi file that includes all of those creds. We're extracting the actual creds using RegExp and JS.

### CODE REVIEW

File: admin/settings/class-admin-apple-settings-section-api.php [hunk 1]
```diff
@@ -85,10 +90,12 @@ public function __construct( $page ) {
...
-			'%s <a target="_blank" href="https://developer.apple.com/news-publisher/">%s</a> %s.',
-			__( 'Enter your Apple News credentials below. See', 'apple-news' ),
-			__( 'the Apple News documentation', 'apple-news' ),
-			__( 'for detailed information', 'apple-news' )
+			'%s <a target="_blank" href="https://developer.apple.com/documentation/apple_news/apple_news_api/getting_ready_to_publish_and_manage_your_articles">%s</a> %s <a target="_blank" href="https://support.apple.com/guide/news-publisher/use-your-cms-with-news-publisher-apd88c8447e6/icloud">%s</a> %s.',
+			__( 'Please upload your Apple News channel configuration file below. Please see', 'apple-news' ),
+			__( 'the Apple News API documentation', 'apple-news' ),
+			__( 'and', 'apple-news' ),
+			__( 'the News Publisher documentation', 'apple-news' ),
+			__( 'for detailed information. For further assistance, please contact your Apple News representative.', 'apple-news' )
```
Comment (kevinfodness) [inline admin/settings/class-admin-apple-settings-section-api.php]: I know this is the same format as it was previously, but really, it should be inverted, like this:
```suggestion
			__( 'Please upload your Apple News channel configuration file below. Please see %sthe Apple News API documentation%s and %sthe News Publisher documentation%s for detailed information. For further assistance, please contact your Apple News representative.', 'apple-news' ),
			'<a target="_blank" href="https://developer.apple.com/documentation/apple_news/apple_news_api/getting_ready_to_publish_and_manage_your_articles">',
			'</a>',
			'<a target="_blank" href="https://support.apple.com/guide/news-publisher/use-your-cms-with-news-publisher-apd88c8447e6/icloud">',
			'</a>',
```

File: assets/js/settings.js [hunk 2]
```diff
@@ -2,6 +2,39 @@
...
+    // storing RegExp strings for decoding the uploaded config file
+    var RegExpStrings = {
+      channel_id: /channel_id: [0-9a-zA-Z_-]*/g,
+      key: /key: [0-9a-zA-Z_-]*/g,
+      secret: /secret: [0-9a-zA-Z_-]*/g
+    }
+
+    function updateCreds(input) {
+      $( '#api_channel' ).val(
+        input.match(RegExpStrings.channel_id).toString().split(" ")[1]
```
Comment (kevinfodness) [inline assets/js/settings.js]: If there is no match, this is going to fail. You can simplify the check by putting the matcher in the regex, like this:

```javascript
/channel_id: ([0-9a-zA-Z_-]+)/
```

And then you can change this to a two-step process, like this:

```javascript
var channelIdMatch = input.match(RegExpStrings.channel_id);
$( '#api_channel' ).val(channelIdMatch ? channelIdMatch[1] : '');
// etc
```

File: assets/js/settings.js [hunk 3]
```diff
@@ -11,5 +44,28 @@
...
+		$( '#api_config_file' ).on( 'change', function (e) {
+			if (!e.target.files || !e.target.files[0]) {
+        return;
+      } else {
+        const file = e.target.files[0];
+        const reader = new FileReader();
+        reader.onload = function(f) {
+          // When a file is uploaded, the read contents populate the hidden textarea.
+          var contents = f.target.result;
+          updateCreds(contents);
```
Comment (kevinfodness) [inline assets/js/settings.js:57]: This is redundant - updating the input's contents should trigger the on change below which triggers `updateCreds`.

**Review by kevinfodness — APPROVED** [review]
A few minor things to clean up, but otherwise looks good. Make sure that you test this thoroughly after making the changes and before merging. 🍣

File: assets/js/settings.js (hunk 3 above)
Comment (tylermachado) [reply assets/js/settings.js:57]: I thought that too, but in testing, that initial FileReader function updated the input textarea but did not fire that subsequent on change, so I had to put in that weird redundancy.

Comment (kevinfodness) [reply assets/js/settings.js:57]: OK, no problem then. Thanks!

**Pull request merged**